src/psyclone/psyir/transformations/loop_tiling_2d_trans.py

```diff
@@ -0,0 +1,150 @@
+        subroutine sub()
+            integer :: ji
+            integer, dimension(100) :: tmp
+            integer :: ji_el_inner
+            integer :: ji_out_var
+            do i_out_var = 1, 100, 32
+              i_el_inner = MIN(i_out_var + (32 - 1), 100)
+              do j_out_var = 1, 100, 32
+                do i = i_out_var, i_el_inner, 1
+                  j_el_inner = MIN(j_out_var + (32 - 1), 100)
+                  do j = j_out_var, j_el_inner, 1
+                    tmp(i, j) = 2 * tmp(i, j)
+                  enddo
+                enddo
+              enddo
+            enddo
+        end subroutine sub
+
+    '''
+    def __str__(self):
+        return "Tile the loop construct using 2D blocks"
+
+    def validate(self, node, options=None):
+        '''
+        Validates that the given Loop node can have a LoopTiling2DTrans
+        applied.
+
+        :param node: the loop to validate.
+        :type node: :py:class:`psyclone.psyir.nodes.Loop`
+        :param options: a dict with options for transformation.
+        :type options: dict of str:values or None
+        :param int options["tilesize"]: The size to tile size for this \
+                transformation. If not specified, the value 32 is used.
+
+        '''
+        if options is None:
+            options = {}
+        super(LoopTiling2DTrans, self).validate(node, options=options)
+        tilesize = options.get("tilesize", 32)
+
+        # Even though the loops that ultimately will be swapped are the ones
+        # resulting from the ChunkLoopTrans, these have the same validation
+        # constrains as swapping the two original loops. This already
+        # guarantees that we have a 2 loop construct with only one loop
+        # statement inside the outer loop.
+        LoopSwapTrans().validate(node)
+
+        # Check that we can chunk both loops
+        outer_loop = node
+        inner_loop = node.loop_body.children[0]
+        ChunkLoopTrans().validate(outer_loop, options={'chuncksize': tilesize})
+        ChunkLoopTrans().validate(inner_loop, options={'chuncksize': tilesize})
+
+    def apply(self, node, options=None):
+        '''
+        Converts the given 2D Loop construct into a tiled version of the nested
+        loops.
+
+        :param node: the loop to transform.
+        :type node: :py:class:`psyclone.psyir.nodes.Loop`
+        :param options: a dict with options for transformations.
+        :type options: dict of str:values or None
+        :param int options["tilesize"]: The size to tile for this \
+                transformation. If not specified, the value 32 is used.
+
+        '''
+        self.validate(node, options)
+        if options is None:
+            options = {}
+        tilesize = options.get("tilesize", 32)
+        parent = node.parent
+        position = node.position
+        outer_loop = node
+        inner_loop = node.loop_body.children[0]
+
+        ChunkLoopTrans().apply(outer_loop, options={'chuncksize': tilesize})
```

Comment:
"chunksize" and again below. Again, implies either a missed test or a missed check?

Reply:
Fixed typo, and added checks and tests in the tiling and the chunking transformations.